File: src/main/java/com/backbase/kalah/service/KalahService.java
package com.backbase.kalah.service;

import com.backbase.kalah.dto.KalahGame;
import com.backbase.kalah.exception.KalahException;

/**
* <h1>KalahService!</h1>
* This interface is responsible for initiating the
* kalah game
*
* @author Gopal Patil
* @version 1.0
* @since   2017-05-18
*/
public interface KalahService {

	public KalahGame init(int stonesPerPit) throws KalahException ;

	public KalahGame validateMove(KalahGame kalahGameRequest) throws KalahException;

	public KalahGame moveStones(KalahGame kalahGameRequest) throws KalahException;
	
}

File: src/main/webapp/resources/js/services/kalah-service.js
/**
 * Created by Gopal Patil
 */
'use strict';

//This service is responsible for sending request to server and get response back.
var AppServices = angular.module('KalahApp.services', []);

AppServices.factory('kalahService', ['$http','$q', function($http, $q) {
	
	return {
		initGame: function(stonesPerPit) {
			return $http.get('/kalah/api/initGame/' + stonesPerPit)
			.then(
					function(response) {
						return response.data;
					}, 
					function(errResponse) {
						console.error('Error while initialising game.');
						return $q.reject(errResponse);
					}
			);
		},
		
		playGame: function(kalahGameRequest) {
			return $http.post('/kalah/api/moveStones', kalahGameRequest)
			.then(
					function(response) {
						return response.data;
					}, 
					function(errResponse) {
						console.error('Error while playing game.');
						return $q.reject(errResponse);
					}
			);
		}
	};
	
}]);